code/obj/item/handheld_dispenser/dispenser.dm

```diff
@@ -0,0 +1,88 @@
+/obj/item/handheld_dispenser
+	name = "handheld pipe dispenser"
+	desc = "A neat tool to quickly lay down pipes onto the floor."
+	icon = 'icons/obj/items/hpd.dmi'
+	icon_state = "hpd"
+	flags = TABLEPASS | CONDUCT
+	var/dispenser_being_used = FALSE
+	var/dispenser_delay = 5 DECI SECONDS
+	var/static/list/atmospipesforcreation = list(
+		"Pipe" = /obj/machinery/atmospherics/pipe/simple/overfloor,
+		"Manifold" = /obj/machinery/atmospherics/pipe/manifold/overfloor,
+		"Quadway manifold" = /obj/machinery/atmospherics/pipe/quadway,
+		"Heat exchanging pipe" = /obj/machinery/atmospherics/pipe/simple/heat_exchanging,
+		"HE junction" = /obj/machinery/atmospherics/pipe/simple/junction,
+	)
+	var/static/list/atmosmachinesforcreation = list(
+		"Passive vent" = /obj/machinery/atmospherics/unary/vent,
+		"Pressure tank" = /obj/machinery/atmospherics/unary/tank,
+		"Passive gate" = /obj/machinery/atmospherics/binary/passive_gate,
+		"Pressure pump" = /obj/machinery/atmospherics/binary/pump,
+		"Volume pump" = /obj/machinery/atmospherics/binary/volume_pump,
+		"Manual valve" = /obj/machinery/atmospherics/binary/valve,
+		"Digital valve" = /obj/machinery/atmospherics/binary/valve/digital,
+		"Outlet Injector" = /obj/machinery/atmospherics/unary/outlet_injector,
+		"Vent pump" = /obj/machinery/atmospherics/unary/vent_pump,
+		"Vent scrubber" = /obj/machinery/atmospherics/unary/vent_scrubber,
+	)
```

Comment:
Would it be better to use defines for these names? Thinking it is going to be confusing remembering the upper/lower casing for these and probably would be better if these names are being used a lot

Reply:
Could actually use `::` syntax to get them directly, but this whole thing needs to be datumized anyway I think